### PR TITLE
test/clean_unit-tests.js

### PR DESCRIPTION
**Pull request by cosycode**
Moved all unit-tests.js into samples/unit-tests and improved some regarding failure handling with help of test-controller.js

### CODE REVIEW

File: samples/unit-tests/chart/getselectedseries/demo.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<script src="https://code.jquery.com/jquery-3.1.1.min.js"></script>
```
Comment (TorsteinHonsi) [inline samples/unit-tests/chart/getselectedseries/demo.html]: Are we actually using jQuery here? This also applies to many other of the new `demo.html` files. I think in the original visual samples we only used jQuery to add button click listeners.

File: samples/unit-tests/exporting/backgroundcolor/demo.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            }],
+
+            navigation: {
+                buttonOptions: {
+                    enabled: false
+                }
+            }
+        });
+
+    // the button handler
```
Comment (TorsteinHonsi) [inline samples/unit-tests/exporting/backgroundcolor/demo.js]: Do we have a button?

File: samples/unit-tests/series/update/demo.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<script src="https://code.highcharts.com/stock/highstock.js"></script>
+<script src="https://code.highcharts.com/highchart.js"></script>
```
Comment (TorsteinHonsi) [inline samples/unit-tests/series/update/demo.html]: Wrong file name => test fails in utils.

File: samples/unit-tests/loading/hideduration/demo.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+
+QUnit.test('Show and hide duration', function (assert) {
```
Comment (TorsteinHonsi) [inline samples/unit-tests/loading/hideduration/demo.js]: This test fails unreliably in the browser (using utils), due to float errors. We should use `QUnit.close` or `toPrecision` on all asserts.

File: samples/unit-tests/chart/reflow/demo.js [hunk 5]
```diff
@@ -1,19 +1,140 @@
 /* eslint func-style:0 */
 
+QUnit.test('Set reflow to false', function (assert) {
```
Comment (TorsteinHonsi) [inline samples/unit-tests/chart/reflow/demo.js]: This test fails for me in Chrome (using utils).

File: samples/unit-tests/series/hide-show-highstock/demo.html [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<script src="https://code.highcharts.com/stock/highstock.js"></script>
```
Comment (TorsteinHonsi) [inline samples/unit-tests/series/hide-show-highstock/demo.html]: Fails in the browser (using utils). Add `usdeur.js`, or use local data instead.

File: samples/unit-tests/series/update/demo.html (hunk 3 above)
Comment (cosycode) [reply samples/unit-tests/series/update/demo.html]: Test does not fails for me. I also do not find any use of Highstock in the demo.js.

File: samples/unit-tests/loading/hideduration/demo.js (hunk 4 above)
Comment (cosycode) [reply samples/unit-tests/loading/hideduration/demo.js]: I will restore the original timing, that has no problem with the float precision.

**Pull request merged**